pallets/loan/src/lib.rs

```diff
@@ -588,13 +721,22 @@ impl<T: Config> Pallet<T> {
 		)
 		.ok_or(Error::<T>::ErrAddAmountOverflow)?;
 
-		LoanInfo::<T>::mutate(pool_id, loan_id, |maybe_loan_info| {
-			let mut loan_info = maybe_loan_info.take().unwrap();
-			loan_info.last_updated = now;
-			loan_info.accumulated_rate = accumulated_rate;
-			loan_info.principal_debt = principal_debt;
-			*maybe_loan_info = Some(loan_info);
-		});
+		LoanInfo::<T>::try_mutate(
+			pool_id,
+			loan_id,
+			|maybe_loan_info| -> Result<(), DispatchError> {
+				let mut loan_data = maybe_loan_info.take().unwrap();
+				loan_data.last_updated = now;
+				loan_data.accumulated_rate = accumulated_rate;
+				loan_data.principal_debt = principal_debt;
+				let new_pv = loan_data
+					.present_value()
+					.ok_or(Error::<T>::ErrLoanPresentValueFailed)?;
+				Self::update_nav_with_updated_present_value(pool_id, new_pv, old_pv)?;
+				*maybe_loan_info = Some(loan_data);
+				Ok(())
+			},
+		)?;
 
 		pallet_pool::Pallet::<T>::repay_currency(
```

Comment:
Can this fail for any reason? If yes, I would move it into the `try_mutate`call.

Reply:
same as above